scripts/Calibration/tofpd/diagnostics.py

```diff
@@ -74,6 +84,36 @@ def _get_difc_ws(wksp, instr_ws=None):
     return difc_ws
 
 
+def __get_regions(x):
+    # Returns a list of tuples with start,stop indices
+    # indicating a detector region
+    regions = []
+    lower = np.asarray((x+1)[:-1], dtype=int)
+    upper = np.asarray((x-1)[1:], dtype=int)
+    gaps = lower<=upper
+    lower, upper = lower[gaps], upper[gaps]
+    i = 0
+    for start, stop in zip(lower, upper):
+        if i == 0:
+            regions.append((int(np.min(x)), int(start-1)))
+        else:
+            regions.append((int(upper[i-1]+1), int(start-1)))
+        i = i + 1
+    # Add ending region if there was atleast one region found:
```

Comment:
typo in `atleast`

Reply:
Also,  comments on the code would help a lot to understand what's being done in this function. I can understand each statement but I don't understand what's the intent